riscv/macros/src/lib.rs

```diff
@@ -265,7 +265,25 @@ impl PacEnumItem {
     }
 
     fn vector_table(&self) -> TokenStream2 {
-        let mut asm = String::from(
+        let mut align = match std::env::var("RISCV_MTVEC_ALIGN") {
+            Ok(x) => x.parse::<u32>().ok(),
+            Err(std::env::VarError::NotPresent) => Some(4),
+            Err(std::env::VarError::NotUnicode(_)) => None,
+        };
+        if let Some(value) = align {
+            if !value.is_power_of_two() || value < 4 {
+                align = None;
+            }
+        }
```

Comment:
```suggestion
        let align = match std::env::var("RISCV_MTVEC_ALIGN") {
            Ok(x) => x.parse::<u32>().ok().and_then(|value| if !value.is_power_of_two() || value < 4 {
                None
            } else {
                Some(value)
            }),
            Err(std::env::VarError::NotPresent) => Some(4),
            Err(std::env::VarError::NotUnicode(_)) => None,
        };
```

Reply:
I did this on the second match for readability.